tools/autograd/templates/VariableType.cpp

```diff
@@ -41,6 +43,52 @@ static void setattr(jit::Node* n, jit::Symbol name, double v)              { n->
 template<std::size_t N>
 static void setattr(jit::Node* n, jit::Symbol name, std::array<bool, N> v) { n->is_(name, std::vector<int64_t>(v.begin(), v.end())); }
 
+template<typename T>
+static jit::Value* createConstant(jit::Node* n, T value) {
+  return n->owningGraph()->createConstant(jit::as_tensor(value))->insertBefore(n)->output();
+}
+
+template<typename T>
+static void genericInsertInput(jit::Node* n, size_t idx, T value) {
+  n->insertInput(idx, createConstant(n, t));
+}
+
+void failPosAttr() {
+  throw std::runtime_error("unsupported type in setposattr. File a bug report!");
+}
+
+static void setposattr(jit::Node* n, size_t idx, const char *name, int64_t v)             { genericInsertInput(n, idx, v); }
+static void setposattr(jit::Node* n, size_t idx, const char *name, const at::Scalar& v)   { genericInsertInput(n, idx, v); }
+static void setposattr(jit::Node* n, size_t idx, const char *name, SparseTensor s)        { failPosAttr(); }
+static void setposattr(jit::Node* n, size_t idx, const char *name, const at::IntList& v)  {
```

Comment:
The code here seems to imply that we'll only "notice" a traced shape when it's inside an IntList. But in principle, shouldn't I be allowed to take an individual scalar from the shape and pass to an argument that expects `int64_t`?

(P.S. I'm guessing how the code works based how I read it; but a Note saying how you are ferrying the traced scalars to the final site would be super helpful.)

Reply:
Yes, it would be possible to expand tracing for `int64_t` inputs as well. It's just that those elements are usually `dim` or some other ordinals, and deriving them from tensor data or shape seems quite uncommon. I decided to keep things simple for now, but in case we'll need this in the future, then all the binding code is already there.